ui/src/app/pages/admin/admin-workspace.tsx

```diff
@@ -39,13 +61,76 @@ const PurpleLabel = ({style = {}, children}) => {
   </label>;
 };
 
+const workspaceInfoField = (labelText: string, divContents: React.ReactFragment) => {
+  return <FlexRow style={styles.infoRow}>
+    <PurpleLabel style={styles.infoLabel}>{labelText}</PurpleLabel>
+    <div style={styles.infoValue}>{divContents}</div>
+  </FlexRow>;
+};
+
+const FileDetailsTable = (props: {data: Array<FileDetail>, bucket: string}) => {
+  const {data, bucket} = props;
+
+  interface TableEntry {
+    name: string;
+    size: string;
+    location: string;
+  }
+
+  const getLocation = (file: FileDetail): string => {
+    const prefix = `${bucket}/`;
+    const suffix = `/${file.name}`;
+    return file.path.replace(prefix, '').replace(suffix, '');
+  };
+
+  const formatMB = (fileSize: number): string => {
```

Comment:
Really nice solution, thank you. Could you apply this to the total on L304 as well?

This also satisfies the AC on https://precisionmedicineinitiative.atlassian.net/browse/RW-5472 , so please close that with this PR as well - if you don't mind.